internal/state.go

```diff
@@ -226,23 +231,20 @@ func PrepareEngineStateWithoutApp(ctx context.Context, headers http.Header) *Sta
 
 //TidyUpEngineURL tidies up an engine url fragment and returns a complete url.
 func TidyUpEngineURL(engine string) string {
-	var url string
-	if strings.HasPrefix(engine, "wss://") {
-		url = engine
-	} else if strings.HasPrefix(engine, "ws://") {
-		url = engine
-	} else {
-		url = "ws://" + engine
-	}
-	if len(strings.Split(url, ":")) == 2 {
-		url += ":9076"
+	if strings.HasPrefix(engine, "wss://") || strings.HasPrefix(engine, "ws://") {
+		return engine
 	}
-	return url
+	return "ws://" + engine
 }
 
 func buildWebSocketURL(engine string, ttl string) string {
 	engine = TidyUpEngineURL(engine)
-	return engine + "/app/engineData/ttl/" + ttl
+	u, _ := neturl.Parse(engine)
+	// Only modify the url if it does not contain a path or ends with a "/"
+	if u.Path == "" && engine[len(engine)-1:] != "/" {
+		engine = engine + "/app/engineData/ttl/" + ttl
```

Comment:
Use `corectl` or something instead of `engineData`?

Reply:
Agree, I will change it to `corectl` instead